### PR TITLE
feat(Select): add ability to persist filtered children state

### PR DESCRIPTION
**Pull request by seanforyou23**
**What**: https://github.com/patternfly/patternfly-react/issues/6300

This PR adds the ability to preserve the filtered list of menu children in Select. Feedback welcome!

Ping @mturley @kmcfaul 

**Additional issues**:  https://github.com/patternfly/patternfly-react/issues/6669

### CODE REVIEW

**Review by mturley — APPROVED** [review]
LGTM! Thanks @seanforyou23

**Comment by mturley** [comment]
@nicolethoen I'm curious about your thoughts here. This originated from the same product issue as https://github.com/patternfly/patternfly-react/issues/6300, and in your comment there (https://github.com/patternfly/patternfly-react/issues/6300#issuecomment-920367065) you were suggesting we use an autocomplete SearchInput instead. I hadn't fully followed up on that, and we decided that doesn't exactly meet the use case here either because we do want to be able to simply click on the field and use it as a Select without typing anything. So I think a prop like this PR introduces makes sense.

The use case here is weird: we want the user to select or create a target namespace for their migration. Many users will only have a few existing ones to choose from, so we don't want to force them to type anything (Select works for that). Some users will have a lot, so the typeahead is good. And the user needs to be able to easily create one (isCreatable), but testers complained when their input was lost if they didn't click the Create option (so isInputValuePersisted makes sense). We then got feedback that it was odd to clear the filtered results if the input wasn't cleared too. This prop seems to complete the puzzle to me (if they want to see all options again they can click the X button).

File: packages/react-core/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -146,6 +146,8 @@ export interface SelectProps
   chipGroupComponent?: React.ReactNode;
   /** Flag for retaining keyboard-entered value in typeahead text field when focus leaves input away */
   isInputValuePersisted?: boolean;
+  /** Flag for retaining filter results on blur from keyboard-entered typeahead text applied with isInputValuePersisted */
```
Comment (nicolethoen) [inline packages/react-core/src/components/Select/Select.tsx:149]: Can you add `@beta` at the beginning of this prop description?

Comment (seanforyou23) [reply packages/react-core/src/components/Select/Select.tsx:149]: Sure no problem

**Comment by nicolethoen** [comment]
I think this should work - marking it beta gives us the flexibility to be sure there are not other side effects of this change.
:) thanks for wiring it up!

**Review by nicolethoen — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]
LGTM!

**Review by tlabaj — APPROVED** [review]

**Pull request merged**